### PR TITLE
Improve Kubernetes access docs

### PR DESCRIPTION
**Pull request by kasperg**
#### What does this PR do?

Improve documenation for accessing Kubernetes.

There are a number of steps/rrerequisites which only needs to be setup once and are thus easy to forget. Let's document them!

#### Should this be tested by the reviewer and how?

Good question. I bet you do not want to reset your entire development environment for this so I guess a sanity check is OK.

### CODE REVIEW

File: documentation/runbooks/access-kubernetes.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ When you need to gain `kubectl` access to a platform-environments Kubernetes clu
 
 * An authenticated `az` cli. The logged in user must have permissions to list
   cluster credentials.
+* `docker` cli which is [authenticated against the GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry).
```
Comment (danquah) [inline documentation/runbooks/access-kubernetes.md:11]: I'm not sure: did we still need this or did the need go away with the update to `.lagoon.yml` to no longer reference the regsitry?

Comment (kasperg) [reply documentation/runbooks/access-kubernetes.md:11]: This is still needed to get [the `dplsh` image](https://github.com/danskernesdigitalebibliotek/dpl-platform/blob/main/tools/dplsh/dplsh.sh#L22).

File: infrastructure/Taskfile.yml [hunk 2]
```diff
@@ -93,17 +99,25 @@ tasks:
...
+        # We normally handle variables via dynamic task variables, but as terraform.
+        # Variables are interpolated before dependencies are resolved, so
+        # in cases where Terraform is not initialized, this task will break.
+        # A lot of other tasks are in the exact same situations.
+        # So, to mitigate this, we instead export the variables we need as
+        # commands.
+        # We sneakily rely on the fact that most procedures starts off with a
+        # cluster:auth, which means most people will have their terraform workspace
+        # initialized before the run the next task which can then extract variables
+        # the "normal" way.
```
Comment (kasperg) [inline infrastructure/Taskfile.yml:115]: :+1: nice.

File: documentation/runbooks/access-kubernetes.md (hunk 1 above)
Comment (danquah) [reply documentation/runbooks/access-kubernetes.md:11]: Ahh, got it. We should probably also add it to https://github.com/danskernesdigitalebibliotek/dpl-platform/blob/main/documentation/runbooks/using-dplsh.md then

**Pull request merged**